src/Symfony/Component/Translation/MessageCatalogueProvider/CacheMessageCatalogueProvider.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\MessageCatalogueProvider;
+
+use Symfony\Component\Config\ConfigCacheInterface;
+use Symfony\Component\Config\ConfigCacheFactoryInterface;
+use Symfony\Component\Config\ConfigCacheFactory;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Manages cache catalogues.
+ *
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class CacheMessageCatalogueProvider implements MessageCatalogueProviderInterface
+{
+    /**
+     * @var MessageCatalogueProviderInterface
+     */
+    private $messageCatalogueProvider;
+
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @var ConfigCacheFactoryInterface
+     */
+    private $configCacheFactory;
+
+    /**
+     * @param MessageCatalogueProviderInterface $messageCatalogueProvider The catalogue bag to use for loading the catalogue.
+     * @param string                            $cacheDir                 The directory to use for the cache.
+     * @param bool                              $debug                    Use cache in debug mode.
+     * @param ConfigCacheFactoryInterface|null  $configCacheFactory       The ConfigCache factory to use.
+     */
+    public function __construct(MessageCatalogueProviderInterface $messageCatalogueProvider, $cacheDir, $debug = false, ConfigCacheFactoryInterface $configCacheFactory = null)
+    {
+        $this->messageCatalogueProvider = $messageCatalogueProvider;
+        $this->cacheDir = $cacheDir;
+        $this->configCacheFactory = $configCacheFactory ?: new ConfigCacheFactory($debug);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCatalogue($locale)
```

Comment:
because I tried to reduce the duplicated code between `Translator` and `MessageCatalogueProvider`, the function [cache()](https://github.com/aitboudad/symfony/blob/translation_loader_resolver/src/Symfony/Component/Translation/Translator.php#L439-L441) is used in the `Translator` too.